### ISSUE TITLE
Add basic support for SQL Targets - [merged]

### ISSUE THREAD

**Issue by MeltyBot** [meltano/sdk#583]
_Merges 263-sql-type-targets-and-sinks -> 74-database-type-streams_

Migrated from GitLab: https://gitlab.com/meltano/sdk/-/merge_requests/200

---

Relates to [#263](https://gitlab.com/meltano/sdk/-/issues/263) 

This MR was built in parallel with SQL Streams.

Remaining tasks to be prioritized: 

- [x] Add tests for modifying tables, adding columns, etc. (`SQLSink.prepare_column()` and `SQLSink._adapt_column_type()`.
- [x] Add record flattening feature - or postpone as follow-on issue. - created as [!236](https://gitlab.com/meltano/sdk/-/merge_requests/236) 
- [ ] Add tests for merge upsert - or migrate to follow-on issue.

Sample implementation for Snowflake in progress here: https://github.com/aaronsteers/target-snowflake/tree/port-to-sql-classes
